core/src/modules/DynamicsModule/include/FreeDriftDynamics.hpp

```diff
@@ -41,17 +41,17 @@ class FreeDriftDynamics : public IDynamics {
         std::cout << tst.start << std::endl;
 
         // set the updated ice thickness and concentration
-        kernel.setData(hiceName, hice.data());
-        kernel.setData(ciceName, cice.data());
+        kernel.setData(hiceName, hice);
```

Comment:
Why `kernel.setData` and `getData` still here?

Reply:
The other dynamics implementations were forgotten. Updated to the new system.